src/backend/utils/misc/queryenvironment.c

```diff
@@ -255,15 +271,31 @@ ENRMetadataGetTupDesc(EphemeralNamedRelationMetadata enrmd)
 bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List **tuplist, int *tuplist_i, int *tuplist_flags)
 {
 	QueryEnvironment *queryEnv = currentQueryEnv;
-	Oid reloid;
 	bool found = false;
 	int index = 0;
 	Datum v1 = 0, v2 = 0, v3 = 0, v4 = 0;
+	Oid pltsql_lang_oid = InvalidOid;
+	Oid pltsql_validator_oid = InvalidOid;
 
-	if (sql_dialect != SQL_DIALECT_TSQL)
-		return false;
+	Oid reloid = RelationGetRelid(rel);
```

Comment:
we should have pltsql_lang_oid check first, followed by reloid check. That way we avoid RelationGetRelid call for APG without BBF.

Reply:
actually `RelationGetRelid` is cheaper and we avoid calling `get_func_language_oids_hook()` for relations other than `SequenceRelationId`

```
#define RelationGetRelid(relation) ((relation)->rd_id)
```